.github/workflows/test-environment-deployment.yml

```diff
@@ -87,6 +84,26 @@ jobs:
           service: ays-be-test-service
           cluster: ays-be-test-cluster
           wait-for-service-stability: true
-
+          
       - name: Deployment successfully completed # Step to indicate that the deployment was successful
-        run: echo "Deployment successfully completed!"
\ No newline at end of file
+        run: echo "Deployment successfully completed!"
+        
+      - name: Cache SonarQube packages # Caching packages related to SonarQube
+        uses: actions/cache@v4 # To use GitHub Actions' cache action
+        with: # The section where parameters are defined for the GitHub Actions action
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar 
+          restore-keys: ${{ runner.os }}-sonar # If the key does not match exactly, the cache will be restored with similar keys.
+          
+      - name: Cache Maven packages # Caching of Maven packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+          
+      - name: Build and analyze # SonarQube integration and Maven execution
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        run: mvn --settings settings.xml -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=afet-yonetim-sistemi_ays-be_d0f7524c-aae5-4394-976c-0b286114d7ff -Dsonar.projectName='ays-be'
```

Comment:
Bu adımları en sona eklemişiz, uygulama deploy olduktan sonra çalışacak. `"Starting Deployment to ECS..."` adımından önce çalışmalı 👀